### PR TITLE
Feature/JS-5824: Implement dots in context date page calendar menu

### PR DESCRIPTION
**Pull request by Nek**
---

- [x] I understand that contributing to this repository will require me to agree with the [CLA](https://github.com/anyproto/.github/blob/main/docs/CLA.md)

---

### What type of PR is this? (check all applicable)

- [ ] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI

### Added tests?

- [ ] 👍 yes
- [ ] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

### Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 [tech-docs](https://github.com/anyproto/tech-docs)
- [ ] 🙅 no documentation needed

### CODE REVIEW

File: src/ts/component/page/main/date.tsx [hunk 1]
```diff
@@ -114,6 +115,59 @@ const PageMainDate = observer(class PageMainDate extends React.Component<I.PageC
...
+
+		U.Data.search({
+			filters: this.getFilters(first, last),
+			keys: [ selectedRelation ],
+		}, (message: any) => {
+			const res = new Map();
+			data.forEach(date => {
+				if (message.records.find(rec => {
+					const recDate = U.Date.getCalendarDateParam(rec[selectedRelation]);
+					return recDate.y == date.y && recDate.m == date.m && recDate.d == date.d;
```
Comment (ra3orblade) [inline src/ts/component/page/main/date.tsx:162]: Parts of logical statements should be wrapped in parenthesis. Also I didn't find declaration for `date` variable.
```
(recDate.y == date.y) && (recDate.m == date.m) && (recDate.d == date.d);
```

File: src/ts/component/menu/dataview/calendar.tsx [hunk 2]
```diff
@@ -102,7 +115,10 @@ const MenuCalendar = observer(class MenuCalendar extends React.Component<I.Menu>
 								}}
 								onContextMenu={e => this.onContextMenu(e, item)}
 							>
-								{item.d}
+								<div className="inner">
+									{item.d}
+									{check && <div className="bullet" /> }
```
[the remark this answers is not in the document]
Comment (Nek) [reply src/ts/component/menu/dataview/calendar.tsx:120]: Why? It's normal for React to write `check && <something />` as it ignores false values by design.

File: src/ts/component/menu/dataview/calendar.tsx [hunk 3]
```diff
@@ -1,19 +1,31 @@
...
+
+const MenuCalendar = observer(class MenuCalendar extends React.Component<I.Menu, State> {
 	
 	originalValue = 0;
 	refMonth: any = null;
 	refYear: any = null;
+
+	state: Readonly<State> = {
+		dotMap: new Map(),
+	}
```
Comment (ra3orblade) [inline src/ts/component/menu/dataview/calendar.tsx:18]: Add semicolon.

File: src/ts/component/menu/dataview/calendar.tsx [hunk 4]
```diff
@@ -1,19 +1,31 @@
 import * as React from 'react';
-import { I, S, U, translate } from 'Lib';
+import { I, S, U, J, translate } from 'Lib';
 import { Select } from 'Component';
 import { observer } from 'mobx-react';
 
-const MenuCalendar = observer(class MenuCalendar extends React.Component<I.Menu> {
+interface State {
+	dotMap: Map<string, boolean>;
+}
```
Comment (ra3orblade) [inline src/ts/component/menu/dataview/calendar.tsx:8]: Add semicolon.

File: src/ts/component/menu/dataview/calendar.tsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (ra3orblade) [reply src/ts/component/menu/dataview/calendar.tsx:121]: It's normal but much harder to read.

File: src/ts/component/page/main/date.tsx [hunk 5]
```diff
@@ -114,6 +115,62 @@ const PageMainDate = observer(class PageMainDate extends React.Component<I.PageC
...
+				quickOption: I.FilterQuickOption.ExactDate,
+				format: I.RelationType.Date,
+			}
+		];
+	};
+
+	getDotMap = (data: {d: number, m: number, y: number}[], callback: (res: Map<string, boolean>) => void): void => {
+		if (data.length < 2) {
+			return;
+		}
```
Comment (ra3orblade) [inline src/ts/component/page/main/date.tsx:150]: Add semicolon.

File: src/ts/component/page/main/date.tsx [hunk 6]
```diff
@@ -114,6 +115,62 @@ const PageMainDate = observer(class PageMainDate extends React.Component<I.PageC
...
+				relationKey: selectedRelation,
+				condition: I.FilterCondition.LessOrEqual,
+				value: end,
+				quickOption: I.FilterQuickOption.ExactDate,
+				format: I.RelationType.Date,
+			}
+		];
+	};
+
+	getDotMap = (data: {d: number, m: number, y: number}[], callback: (res: Map<string, boolean>) => void): void => {
```
Comment (ra3orblade) [inline src/ts/component/page/main/date.tsx:147]: callback everywhere is written in camelCase, `callBack`

**Review by ra3orblade — CHANGES_REQUESTED** [review]

**Review by ra3orblade — APPROVED** [review]

**Pull request merged**